### PR TITLE
Fill out NatSpec comments

### PR DESCRIPTION
**Pull request by ThunderDeliverer**
This is the first PR that fills out our smart contract with the NatSpec comments in order to generate  the full documentation of our smart contract suite.

The `dodoc` dependency was added to `Hardhat` in order to generate the docs from the comments.

### CODE REVIEW

**Comment by ThunderDeliverer** [comment]
I know this PR looks daunting, but most of the changed files are autogenerated (the whole `docs/` directory), so it's not as bad as it looks 😬

File: contracts/RMRK/access/OwnableLock.sol [hunk 1]
```diff
@@ -19,96 +22,113 @@ contract OwnableLock is Context {
...
+     * @notice Locks the operation.
+     * @dev Once locked, functions using `notLocked` modifier cannot be executed.
      */
     function setLock() external onlyOwner {
         _lock = true;
     }
 
     /**
-     * @dev Returns lock status.
+     * @notice Reenables the operation of functions using `notLocked` modifier.
```
Comment (ThunderDeliverer) [inline contracts/RMRK/access/OwnableLock.sol:71]: This is wrong.

```suggestion
     * @notice Used to retrieve the status of a lockable smart contract.
     * @return bool A boolean value signifying whether the smart contract has been locked (`true`) or not (`false`)
```

**Review by steven2308 — APPROVED** [review]
Just added comment, but will add a FIXME instead.
I found a few things to improve but change them directly in another commit.

File: contracts/RMRK/base/IRMRKBaseStorage.sol [hunk 2]
```diff
@@ -45,11 +65,16 @@ interface IRMRKBaseStorage is IERC165 {
...
-    @dev Base struct for a standard RMRK base item. Requires a minimum of 3 storage slots per base item,
-    * equivalent to roughly 60,000 gas as of Berlin hard fork (April 14, 2021), though 5-7 storage slots
-    * is more realistic, given the standard length of an IPFS URI. This will result in between 25,000,000
-    * and 35,000,000 gas per 250 resources--the maximum block size of ETH mainnet is 30M at peak usage.
-    */
+     * @notice The integral structure of a standard RMRK base item defining it.
+     * @dev Requires a minimum of 3 storage slots per base item, equivalent to roughly 60,000 gas as of Berlin hard fork
+     *  (April 14, 2021), though 5-7 storage slots is more realistic, given the standard length of an IPFS URI. This
+     *  will result in between 25,000,000 and 35,000,000 gas per 250 resources--the maximum block size of Ethereum
+     *  mainnet is 30M at peak usage.
```
[the remark this answers is not in the document]
Comment (ThunderDeliverer) [reply contracts/RMRK/base/IRMRKBaseStorage.sol:72]: The block gas limit is still 30M gas, so I think so.

[the remark this answers is not in the document]
Comment (steven2308) [reply contracts/RMRK/base/IRMRKBaseStorage.sol:72]: I mean the storage use.  IDK if the struct changed since this comment was added

**Pull request merged**